### PR TITLE
Added stricter validation on some new titus objects

### PR DESCRIPTION
**Pull request by solarkennedy**
I was sprinkling in @Valid, not knowing what it really does.
From reading the docs, it sounds like it is designed to
operate on an *Object* and enforce recursive validation.

This PR puts more specific validation on strings and such.

### CODE REVIEW

**Comment by tbak** [comment]
@solarkennedy It would be great if you could add some tests in `JobModelSanitizationTest` to check these new rules.

**Comment by solarkennedy** [comment]
Whoa, a real failure:
```
        java.lang.RuntimeException: GRPC status Status{code=INVALID_ARGUMENT, description=TitusServiceException: Invalid Argument: {Validation failed: 'field: 'platformSidecars[0].channel', description: 'channel names name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character', type: 'HARD''}, {Validation failed: 'field: 'platformSidecars[0].name', description: 'platform sidecar names name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character', type: 'HARD''}, cause=null}: TitusServiceException: Invalid Argument: {Validation failed: 'field: 'platformSidecars[0].channel', description: 'channel names name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character', type: 'HARD''}, {Validation failed: 'field: 'platformSidecars[0].name', description: 'platform sidecar names name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character', type: 'HARD''}
```

**Review by andrew-leung — APPROVED** [review]

File: titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/PlatformSidecar.java [hunk 1]
```diff
@@ -2,17 +2,20 @@
...
 import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
 
 
 public class PlatformSidecar {
 
-    @Valid
+    @NotNull
+    @Pattern(regexp = "[a-z0-9]([-a-z0-9]*[a-z0-9])?", message = "platform sidecar names name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character")
```
Comment (aaronlehmann) [inline titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/PlatformSidecar.java:12]: Let's make these regexps consistent with the validation we apply at the InfraAPI level: https://stash.corp.netflix.com/projects/NCE/repos/platform-sidecar-validator/browse/validate.go#231

Comment (solarkennedy) [reply titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/PlatformSidecar.java]: added

**Review by tbak — APPROVED** [review]

**Pull request merged**